src/builder/create_components.rs

```diff
@@ -47,6 +56,462 @@ impl serde::Serialize for CreateActionRow<'_> {
     }
 }
 
+/// A builder for creating components in a structured way.
+///
+/// This enum supports both V1 and V2 components, with the exception of `ActionRow`, which is a V1
+/// component.
+///
+/// ## V2 Components and Message Flags
+/// To send V2 components, you must set [`MessageFlags::IS_COMPONENTS_V2`].
+///
+/// ### Limitations
+/// - The total number of components is limited to **40**..
+/// - The maximum character count for text within components is **4000**.
+/// - The ability to set the `content` and `embeds` field will be disabled
+/// - No support for audio files
+/// - No simple text preview for files
+/// - No embeds for urls
+#[derive(Clone, Debug, Serialize)]
+#[must_use]
+#[serde(untagged)]
+pub enum CreateComponent<'a> {
+    /// Represents an action row component (V1).
+    ///
+    /// An action row is a container for other interactive components, such as buttons and select
+    /// menus.
+    ActionRow(CreateActionRow<'a>),
+    /// Represents a section component (V2).
+    ///
+    /// A section is used to structure and group other components with an accessory.
+    Section(CreateSection<'a>),
+    /// Represents a text display component (V2).
+    ///
+    /// This component is used for displaying text within a message, separate from interactive
+    /// elements.
+    TextDisplay(CreateTextDisplay<'a>),
+    /// Represents a media gallery component (V2).
+    ///
+    /// A media gallery allows embedding images, videos, or other media assets within a message.
+    MediaGallery(CreateMediaGallery<'a>),
+    /// Represents a file component (V2).
+    ///
+    /// This component is used for attaching and displaying files within a message.
+    File(CreateFile<'a>),
+    /// Represents a separator component (V2).
+    ///
+    /// A separator is used to visually divide sections within a message for better readability.
+    Separator(CreateSeparator),
+    /// Represents a container component (V2).
+    ///
+    /// A container is a flexible component that can hold multiple nested components.
+    Container(CreateContainer<'a>),
+}
+
+/// A builder to create a section component, supports up to a max of **3** components with an
+/// accessory.
+#[derive(Clone, Debug, Serialize)]
+#[must_use]
+pub struct CreateSection<'a> {
+    #[serde(rename = "type")]
+    kind: StaticU8<9>,
```

Comment:
This may have been intentionally omitted, but every component type is documented to have an optional integer `id` field (documented as being limited to 32-bits). I'll just note this once here, but this affects every `Create*` for a component, as well as the "received" versions in `model/application/component.rs`.